lib/capybara/selector/definition/button.rb

```diff
@@ -19,17 +19,19 @@
 
       input_btn_xpath = input_btn_xpath[locator_matchers] + locate_label(locator).descendant(input_btn_xpath)
 
-      btn_xpath = btn_xpath[locator_matchers |
-                            XPath.string.n.is(locator) |
-                            XPath.descendant(:img)[XPath.attr(:alt).is(locator)]
-                           ] + locate_label(locator).descendant(btn_xpath)
+      btn_xpaths = btn_xpaths.map do |xpath|
+        xpath[locator_matchers |
+              XPath.string.n.is(locator) |
+              XPath.descendant(:img)[XPath.attr(:alt).is(locator)]
+             ] + locate_label(locator).descendant(xpath)
+      end
```

Comment:
@twalpole mapping over the `btn_xpaths` instead of union-ing them from the start seems to have done the trick. Unfortunately, this loop and other changes has pushed this block over the rubocop limit. 

What techniques are at our disposal for refactoring long blocks in these DSLs?